### ISSUE TITLE
Error when using Quant_bins

### ISSUE THREAD

**Issue by Timmywang1** [bxlab/metaWRAP#280]
metawrap quant_bins -b /share/home/binhaow/metagenomedata3/BIN_REFINEMENT3/metawrap_60_10_bins/ -t 20 -o /share/home/binhaow/metagenomedata4/QUANT_BINS3 -a /share/home/binhaow/metagenomedata3/ASSEMBLY3/megahit/final.contigs1_5.fa /share/home/binhaow/metagenomedata3/CLEAN_READS3/clean*.fastq

-----------------------------------------------------------------------------------------------------------------------
-----                                           summarize salmon files...                                          -----
------------------------------------------------------------------------------------------------------------------------

Starting in: /share/home/binhaow/metagenomedata4/QUANT_BINS3/alignment_files
Loading counts from: ./clean.S1.quant quant.sf
Loading counts from: ./clean.S2.quant quant.sf
Loading counts from: ./clean.S3.quant quant.sf
Loading counts from: ./clean.S4.quant quant.sf
Loading counts from: ./clean.S5.quant quant.sf
"clean.S1.quant.counts",
"clean.S2.quant.counts",
"clean.S3.quant.counts",
"clean.S4.quant.counts",
"clean.S5.quant.counts"

########################################################################################################################
########################################################################################################################

------------------------------------------------------------------------------------------------------------------------
-----                            There were 5 samples detected. Making abundance table!                            -----
------------------------------------------------------------------------------------------------------------------------

************************************************************************************************************************
*****                     something went wrong with making summary abundance table. Exiting...                     *****
************************************************************************************************************************

real	135m7.227s
user	454m45.277s
sys	44m59.610s

**Comment by ursky** [comment]
Can you provide the contents of `clean.S1.quant.counts` and the other files? Its strange there is no stderr merrage,,,

**Comment by Timmywang1** [comment]
head -n 40 clean.S1.quant.counts
transcript	count
k141_1	0.0
k141_2	0.0
k141_3	0.474976
k141_4	0.0
k141_5	0.0
k141_6	0.0
k141_7	0.378531
k141_8	0.122853
k141_9	0.0
k141_10	0.0
k141_11	0.0
k141_12	0.0
k141_13	0.287786
k141_14	0.0
k141_15	0.0
k141_16	0.622564
k141_17	0.344229
k141_18	0.355191
k141_19	0.098537
k141_20	0.357389
k141_21	0.449514
k141_23	0.073421
k141_24	0.172549
k141_25	1.366322
k141_26	0.17674
k141_27	0.0
k141_28	0.0
k141_29	0.0
k141_30	0.0
k141_31	0.633335
k141_32	0.0
k141_33	0.0
k141_34	0.129662
k141_35	0.470713
k141_36	0.0
k141_37	0.383072
k141_38	0.355191
k141_39	0.216108
k141_40	0.490431

Thank you for your help!

**Comment by ursky** [comment]
Can you go in and investidate what the exact component is that failed? This is the exact script that is failing in the pipeline: 
`
bin/metawrap-scripts/split_salmon_out_into_bins.py /share/home/binhaow/metagenomedata4/QUANT_BINS3/quant_files/ /share/home/binhaow/metagenomedata3/BIN_REFINEMENT3/metawrap_60_10_bins/ /share/home/binhaow/metagenomedata3/ASSEMBLY3/megahit/final.contigs1_5.fa > /share/home/binhaow/metagenomedata4/QUANT_BINS3/bin_abundance_table.tab
`

Basically, it takes in the contents of `QUANT_BINS3/quant_files/`. Can you check if anything looks suspicious? And what error the command throws?

**Comment by Timmywang1** [comment]
When I run the command--"bin/metawrap-scripts/split_salmon_out_into_bins.py ", the contents in the table are as follows:
None of the contigs/scaffolds in the -a metagenomic assembly file were present in the bin files. Please make sure that the bins and total assembly have the exact same bins. One cause for this could be that you reassembled the bins, disrupring the contig naming. If you do not have the original total metagenomic assembly file, then you could not provide the -a option at all (but this is not ideal for abundance estimation).

**Comment by ursky** [comment]
Ah. There's your answer right there. The contigs in the assembly you provided did not match those in the bins you provided. Double-check your inputs by looking at the contig names, and re-run once you find what was wrong. You could also just not provide the assembly, which would make the results slightly less accurate.

**Comment by Timmywang1** [comment]
hello, I have some problems. I have 5 samples. I don’t want to know the average abundance. I want to know the abundance in each sample. How should I set it?
<metawrap quant_bins -b metawrap_60_10_bins/ -t 20 -o QUANT_BINS3 -a final_assembly.fasta CLEAN_READS3/clean.S1_1.fastq CLEAN_READS3/clean.S1_2.fastq CLEAN_READS3/clean.S2_1.fastq CLEAN_READS3/clean.S2_2.fastq CLEAN_READS3/clean.S3_1.fastq CLEAN_READS3/clean.S3_2.fastq CLEAN_READS3/clean.S4_1.fastq CLEAN_READS3/clean.S4_2.fastq CLEAN_READS3/clean.S5_1.fastq CLEAN_READS3/clean.S5_2.fastq>

------------------------------------------------------------------------------------------------------------------------
-----                            There were 5 samples detected. Making abundance table!                            -----
------------------------------------------------------------------------------------------------------------------------


------------------------------------------------------------------------------------------------------------------------
-----                     Average bin abundance table stored in QUANT_BINS3/abundance_table.tab                    -----
------------------------------------------------------------------------------------------------------------------------


########################################################################################################################
#####                                 MAKING GENOME ABUNDANCE HEATMAP WITH SEABORN                                 #####
########################################################################################################################


------------------------------------------------------------------------------------------------------------------------
-----                                          making heatmap with Seaborn                                         -----
------------------------------------------------------------------------------------------------------------------------

loading libs...
loading abundance data...
drawing clustermap...
Traceback (most recent call last):
  File "/share/home/binhaow/anaconda2/envs/metawrap/bin/metawrap-scripts/make_heatmap.py", line 75, in <module>
    draw_clustermap(df, lut)
  File "/share/home/binhaow/anaconda2/envs/metawrap/bin/metawrap-scripts/make_heatmap.py", line 58, in draw_clustermap
    g = sns.clustermap(df, figsize=(14,8), col_cluster=True, yticklabels=True, cmap="magma")
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/seaborn/matrix.py", line 1301, in clustermap
    **kwargs)
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/seaborn/matrix.py", line 1131, in plot
    row_linkage=row_linkage, col_linkage=col_linkage)
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/seaborn/matrix.py", line 1032, in plot_dendrograms
    axis=1, ax=self.ax_col_dendrogram, linkage=col_linkage)
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/seaborn/matrix.py", line 746, in dendrogram
    label=label, rotate=rotate)
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/seaborn/matrix.py", line 564, in __init__
    self.linkage = self.calculated_linkage
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/seaborn/matrix.py", line 628, in calculated_linkage
    return self._calculate_linkage_scipy()
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/seaborn/matrix.py", line 603, in _calculate_linkage_scipy
    metric=self.metric)
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/scipy/cluster/hierarchy.py", line 716, in linkage
    n = int(distance.num_obs_y(y))
  File "/share/home/binhaow/anaconda2/envs/metawrap/lib/python2.7/site-packages/scipy/spatial/distance.py", line 2276, in num_obs_y
    raise ValueError("The number of observations cannot be determined on "
ValueError: The number of observations cannot be determined on an empty distance matrix.

************************************************************************************************************************
*****                           something went wrong with making the heatmap. Exiting...                           *****
************************************************************************************************************************

**Comment by ursky** [comment]
It would be helpful to have the entire output (strout + stderr) to daignose this.
